server/src/main/java/org/candlepin/model/OwnerProductCurator.java

```diff
@@ -0,0 +1,400 @@
+            .uniqueResult();
+    }
+
+    public Collection<Owner> getOwnersByProduct(Product product) {
+        return this.getOwnersByProduct(product.getId());
+    }
+
+    @Transactional
+    public Collection<Owner> getOwnersByProduct(String productId) {
+        return (List<Owner>) this.createSecureCriteria()
+            .createAlias("product", "product")
+            .setProjection(Projections.property("owner"))
+            .add(Restrictions.eq("product.id", productId))
+            .list();
+    }
+
+    public Collection<Product> getProductsByOwner(Owner owner) {
+        return this.getProductsByOwner(owner.getId());
+    }
+
+    @Transactional
+    public Collection<Product> getProductsByOwner(String ownerId) {
+        return (List<Product>) this.createSecureCriteria()
+            .createAlias("owner", "owner")
+            .setProjection(Projections.property("product"))
+            .add(Restrictions.eq("owner.id", ownerId))
+            .list();
+    }
+
+    public Collection<Product> getProductsByIds(Owner owner, Collection<String> productIds) {
+        return this.getProductsByIds(owner.getId(), productIds);
+    }
+
+    @Transactional
+    public Collection<Product> getProductsByIds(String ownerId, Collection<String> productIds) {
+        Collection<Product> result = null;
+
+        if (productIds != null && productIds.size() > 0) {
+            Criteria criteria = this.createSecureCriteria()
+                .createAlias("owner", "owner")
+                .createAlias("product", "product")
+                .setProjection(Projections.property("product"))
+                .add(Restrictions.eq("owner.id", ownerId))
+                .add(this.unboundedInCriterion("product.id", productIds));
+
+            result = (Collection<Product>) criteria.list();
+        }
+
+        return result != null ? result : new LinkedList<Product>();
+    }
+
+    @Transactional
+    public long getOwnerCount(Product product) {
+        String jpql = "SELECT count(op) FROM OwnerProduct op WHERE op.product.uuid = :product_uuid";
+
+        long count = (Long) this.getEntityManager()
+            .createQuery(jpql, Long.class)
+            .setParameter("product_uuid", product.getUuid())
+            .getSingleResult();
+
+        return count;
+    }
+
+    @Transactional
+    public boolean isProductMappedToOwner(Product product, Owner owner) {
+        String jpql = "SELECT count(op) FROM OwnerProduct op " +
+            "WHERE op.owner.id = :owner_id AND op.product.uuid = :product_uuid";
+
+        long count = (Long) this.getEntityManager()
+            .createQuery(jpql)
+            .setParameter("owner_id", owner.getId())
+            .setParameter("product_uuid", product.getUuid())
+            .getSingleResult();
+
+        return count > 0;
+    }
+
+    @Transactional
+    public boolean mapProductToOwner(Product product, Owner owner) {
+        if (!this.isProductMappedToOwner(product, owner)) {
```

Comment:
Catching the exception looks to be off the table. Even caught, Hibernate marks the session as rollback only, so we can't commit the change.